src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1979,182 +1987,194 @@ BuildJobTreeTaskList(Job *jobTree)
 
 /*
  * SubquerySqlTaskList creates a list of SQL tasks to execute the given subquery
- * pushdown job. For this, it gets all range tables in the subquery tree, then
- * walks over each range table in the list, gets shards for each range table,
- * and prunes unneeded shards. Then for remaining shards, fragments are created
- * and merged to create fragment combinations. For each created combination, the
- * function builds a SQL task, and appends this task to a task list.
+ * pushdown job. For this, the it is being checked whether the query is router
+ * plannable per target shard interval. For those router plannable worker
+ * queries, we create a SQL task and append the task to the task list that is going
+ * to be executed.
  */
 static List *
-SubquerySqlTaskList(Job *job)
+SubquerySqlTaskList(Job *job, PlannerRestrictionContext *plannerRestrictionContext)
 {
 	Query *subquery = job->jobQuery;
 	uint64 jobId = job->jobId;
 	List *sqlTaskList = NIL;
-	List *fragmentCombinationList = NIL;
-	List *opExpressionList = NIL;
-	List *queryList = NIL;
 	List *rangeTableList = NIL;
-	ListCell *fragmentCombinationCell = NULL;
 	ListCell *rangeTableCell = NULL;
-	ListCell *queryCell = NULL;
-	Node *whereClauseTree = NULL;
 	uint32 taskIdIndex = 1; /* 0 is reserved for invalid taskId */
-	uint32 anchorRangeTableId = 0;
-	uint32 rangeTableIndex = 0;
-	const uint32 fragmentSize = sizeof(RangeTableFragment);
-	uint64 largestTableSize = 0;
+	Oid relationId = 0;
+	int shardCount = 0;
+	int shardOffset = 0;
+	DistTableCacheEntry *targetCacheEntry = NULL;
+	RelationRestrictionContext *relationRestrictionContext =
+		plannerRestrictionContext->relationRestrictionContext;
+	bool allRelationsJoinedOnPartitionKey PG_USED_FOR_ASSERTS_ONLY =
```

Comment:
I think it is worth keeping this `Assert` here. Because all the assumptions made in this function is based on the result of that function in the logical planner. So, I think it may prevent any future changes to break the assumption here.